src/families/ethereum/modules/send.js

```diff
@@ -99,13 +106,60 @@ const send: ModeModule = {
     }
   },
 
-  fillDeviceTransactionConfig({ status: { amount } }, fields) {
+  fillDeviceTransactionConfig(
+    { transaction, account, status: { amount } },
+    fields
+  ) {
     if (!amount.isZero()) {
       fields.push({
         type: "amount",
         label: "Amount",
       });
     }
+    if (transaction.data?.length) {
+      const token = findTokenByAddress(transaction.recipient);
+      // $FlowFixMe (transaction data is not null, you flow)
+      const method = transaction.data.slice(0, 4).toString("hex");
+
+      if (method === "095ea7b3" && token) {
+        fields.push({
+          type: "text",
+          label: "Type",
+          value: `Approve`,
+        });
+        fields.push({
+          type: "text",
+          label: "Amount (1/2)",
+          value: token.ticker,
+        });
+
+        // $FlowFixMe (transaction data is not null, you flow)
+        const amountHex = transaction.data.slice(36, 36 + 32).toString("hex");
+
+        if (uniq(amountHex.split("")) === ["f"]) {
```

Comment:
i'm not confident for us to maintain this ourself. we never did the "Amount (1/2)" display on Live side at the moment and i don't think we need to start doing this unless if it's addressed very generically.